pootle/apps/pootle_store/models.py

```diff
@@ -2033,11 +2038,15 @@ def get_max_unit_revision(self):
     def can_be_updated(self):
         return not self.obsolete
 
-    def get_parent(self):
+    def get_parents(self):
         if self.parent.is_translationproject():
-            return self.translation_project
+            parents = [self.translation_project]
         else:
-            return self.parent
+            parents = [self.parent]
+
+        parents.extend(self.parent_vf_treeitems.all())
```

Comment:
I think we should skip DB querying for `parent_vf_treeitems.all()` if we don't use virtual folders.